cves/CVE-2009-2693.yml

```diff
@@ -1,205 +1,216 @@
+    directory names. Look at comments in the code. Look at the bug reports how
+    the bug report was tagged.
+    Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
+  answer: startup
+  name: startup
+
+interesting_commits:
+  question: |
+    Are there any interesting commits between your VCC(s) and fix(es)?
+    Write a brief (under 100 words) description of why you think this commit was
+    interesting in light of the lessons learned from this vulnerability. Any
+    emerging themes?
+    If there are no interesting commits, demonstrate that you completed this
+    section by explaining what happened between the VCCs and the fix.
+  answer:
+  commits:
+    - commit: 3b0ea850738c091299373a330929cc024dd1867a
+      note: There isn't anything interesting happening. But this commit reverted the
+        previous commit since the developer was not satisfied with the fix. 
+
+lessons:
+  question: |
+    Are there any common lessons we have learned from class that apply to this
+    vulnerability? In other words, could this vulnerability serve as an example
+    of one of those lessons?
+    Leave "applies" blank or put false if you did not see that lesson (you do
+    not need to put a reason). Put "true" if you feel the lesson applies and put
+    a quick explanation of how it applies.
+    Don't feel the need to claim that ALL of these apply, but it's pretty likely
+    that one or two of them apply.
+    If you think of another lesson we covered in class that applies here, feel
+    free to give it a small name and add one in the same format as these.
+  path_traversal:
+    applies: true
+    note: This is directly an example of using ../ to navigate outside the root directory
+  defense_in_depth:
+    applies: false
+    note:
+  least_privilege:
+    applies: false
+    note:
+  frameworks_are_optional:
+    applies: false
+    note:
+  native_wrappers:
+    applies: false
+    note:
+  distrust_input:
+    applies: false
+    note:
+  security_by_obscurity:
+    applies: false
+    note:
+  serial_killer:
+    applies: false
+    note:
+  environment_variables:
+    applies: false
+    note:
+  secure_by_default:
+    applies: false
+    note:
+  yagni:
+    applies: false
+    note:
+  complex_inputs:
+    applies: false
+    note: 
+
+mistakes:
+  question: |
+    In your opinion, after all of this research, what mistakes were made that
+    led to this vulnerability? Coding mistakes? Design mistakes?
+    Maintainability? Requirements? Miscommunications?
+    Look at the CWE entry for this vulnerability and examine the mitigations
+    they have written there. Are they doing those? Does the fix look proper?
+    Use those questions to inspire your answer. Don't feel obligated to answer
+    every one. Write a thoughtful entry here that those ing the software
+    engineering industry would find interesting.
+  answer: The mistake that was made here was simple. The developer neglected to consider

```

Comment:
may want to mention the lack of unit tests around this prior to the vulnerability being discovered.

Reply:
I added a mention of lack of unit testing. Good idea!